agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -122,7 +122,10 @@ def ensure_service_worker(service_pod)
           workers[service_pod.id].async.update(service_pod)
         end
       rescue Celluloid::DeadActorError
+        error "tried to call dead worker: #{service_pod.name}"
         workers.delete(service_pod.id)
+        info "recreating worker for: #{service_pod.name}"
+        ensure_service_worker(service_pod)
```

Comment:
I suspect this isn't the right place to have this logic... it only specifically applies to `Celluloid::DeadActorError`s raised from `async.update` calls .... with the actor linking, a crash in the async `update` call would first show up as a `on_worker_exit`, which removes it from the workers?

* the `ServicePodWorker` crashes in the `on_container_event` immediately after the `initialize` but before the manager calls `async.update`
* the `ServicePodWorker` crashes in the `async.update`, but before celluloid is able to propagate the linked actor crash to the manager `trap_exit :on_worker_exit`, the manager `populate_workers_from_master` loop calls `ensure_service_worker`

I don't exactly see when this specific rescue would ever happen, apart from some very rare race condition scenarios?